pkg/pillar/agentlog/agentlog.go

```diff
@@ -225,9 +226,22 @@ func writeOrLog(log *base.LogObject, w io.Writer, msg string) {
 	}
 }
 
-var listenDebugRunning atomic.Bool
+var (
+	listenDebugRunning  atomic.Bool
+	psiCollectorRunning atomic.Bool
+	psiCollectorCtx     context.Context
+	psiCollectorCancel  context.CancelFunc
+)
 
-func listenDebug(log *base.LogObject, stacksDumpFileName, memDumpFileName string) {
+// ListenDebug starts a debug server on localhost:6543
+// It exposes the net/http/pprof API
+// It also allows to start and stop the memory PSI collector by sending a POST request to
+// /memory-monitor/psi-collector/start and /memory-monitor/psi-collector/stop
+// The server can be stopped by sending a POST request to /stop
+// The server can be used to dump stacks by sending a POST request to /dump/stacks
+// The server can be used to dump memory info by sending a POST request to /dump/memory
```

Comment:
perhaps put this comment into https://github.com/lf-edge/eve/pull/4141/files#diff-e76641ce9bfa02962b0687e68a963a44dfa9d1a4f427cfafd9fbfa36d55986f4L249 ?

Reply:
will do

Reply:
fixed